CrowdsaleToken.sol

```diff
@@ -0,0 +1,216 @@
+   * and the new allowance by unfortunate transaction ordering. One possible solution to mitigate this
+   * race condition is to first reduce the spender's allowance to 0 and set the desired value afterwards:
+   * https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
+   * @param _spender The address which will spend the funds.
+   * @param _value The amount of tokens to be spent.
+   */
+  function approve(address _spender, uint256 _value) public returns (bool) {
+    allowed[msg.sender][_spender] = _value;
+    Approval(msg.sender, _spender, _value);
+    return true;
+  }
+
+  /**
+   * @dev Function to check the amount of tokens that an owner allowed to a spender.
+   * @param _owner address The address which owns the funds.
+   * @param _spender address The address which will spend the funds.
+   * @return A uint256 specifying the amount of tokens still available for the spender.
+   */
+  function allowance(address _owner, address _spender) public constant returns (uint256 remaining) {
+    return allowed[_owner][_spender];
+  }
+
+  /**
+   * approve should be called when allowed[_spender] == 0. To increment
+   * allowed value is better to use this function to avoid 2 calls (and wait until
+   * the first transaction is mined)
+   * From MonolithDAO Token.sol
+   */
+  function increaseApproval (address _spender, uint _addedValue)
+    returns (bool success) {
+    allowed[msg.sender][_spender] = allowed[msg.sender][_spender].add(_addedValue);
+    Approval(msg.sender, _spender, allowed[msg.sender][_spender]);
+    return true;
+  }
+
+  function decreaseApproval (address _spender, uint _subtractedValue)
+    returns (bool success) {
+    uint oldValue = allowed[msg.sender][_spender];
+    if (_subtractedValue > oldValue) {
+      allowed[msg.sender][_spender] = 0;
+    } else {
+      allowed[msg.sender][_spender] = oldValue.sub(_subtractedValue);
+    }
+    Approval(msg.sender, _spender, allowed[msg.sender][_spender]);
+    return true;
+  }
+
+
+}
+
+
+/*  CROWDSALE TOKEN:
+    Have uniform price throughout the sell.
+    Immediately issue the tokens to the buyers.
+    Sends the etheres to the owner (deployer) of the ICO contract (only one address).
+    
+    Following Crowdsale Token at :
+    http://zeltsinger.com/2017/04/22/ico-simple-simple/
+    
+*/
+
+contract CrowdsaleToken is StandardToken {
+    
+    using SafeMath for uint;
+    
+    string public constant name = "AkayCoin";
+    string public constant symbol = "AK";
+    uint8 public constant decimals = 18;
+    
+    // address where funds are collected
+    address public constant wallet = 0xed8862bcb2a3186f6123ed924dbb4cfc5a747491;
+    
+    //1 ether = 500 example tokens
+    uint public constant PRICE = 500;
+    
+    
+    function () payable {
+        createTokens(msg.sender);
+    }
+    
+    function createTokens(address recipient) payable {
```

Comment:
So this is mintable crowd sale. It is very rare you will find a chance to launch mintable crowdsale. In most of the cases you will given a fixed total supply and token price